src/app/ReadHandler.h

```diff
@@ -83,22 +89,96 @@ class ReadHandler : public Messaging::ExchangeDelegate
         virtual void OnDone(ReadHandler & apReadHandlerObj) = 0;
     };
 
+    /*
+     * A callback used to interact with the application.
+     */
+    class ApplicationCallback
+    {
+    public:
+        virtual ~ApplicationCallback() = default;
+
+        /*
+         * Called right after a SubscribeRequest has been parsed and processed. This notifies an interested application
+         * of a subscription that is about to be established. It also provides an avenue for altering the parameters of the
+         * subscription (specifically, the min/max negotiated intervals) or even outright rejecting the subscription for
+         * application-specific reasons.
+         *
+         * TODO: Need a new IM status code to convey application-rejected subscribes. Currently, an OUT_OF_RESOURCE error is sent
```

Comment:
This does not match the actual error the code returns.

Reply:
But what's sending the OUT_OF_RESOURCE status?

Reply:
Apologies, it returns `Protocols::InteractionModel::Status::Failure` (set in InteractionModelEngine.cpp:428).